microceph/ceph/rgw.go

```diff
@@ -3,25 +3,52 @@ package ceph
 import (
 	"context"
 	"database/sql"
+	"encoding/base64"
 	"fmt"
+	"github.com/canonical/microceph/microceph/constants"
+	"github.com/canonical/microceph/microceph/interfaces"
 	"os"
 	"path/filepath"
 	"strings"
 
-	"github.com/canonical/microceph/microceph/constants"
-	"github.com/canonical/microceph/microceph/interfaces"
-
 	"github.com/canonical/microceph/microceph/database"
 )
 
 // EnableRGW enables the RGW service on the cluster and adds initial configuration given a service port number.
-func EnableRGW(s interfaces.StateInterface, port int, monitors []string) error {
+func EnableRGW(s interfaces.StateInterface, port int, sslPort int, sslCertificate string, sslPrivateKey string, monitors []string) error {
 	pathConsts := constants.GetPathConst()
 
+	sslCertificatePath := ""
+	sslPrivateKeyPath := ""
+	if sslCertificate != "" && sslPrivateKey != "" {
+		sslCertificatePath = filepath.Join(pathConsts.SSLFilesPath, "server.crt")
+		decodedSSLCertificate, err := base64.StdEncoding.DecodeString(sslCertificate)
+		if err != nil {
+			return err
+		}
+		err = writeFile(sslCertificatePath, string(decodedSSLCertificate), 0755)
```

Comment:
Hm, this writes the files with the eXecutable bit set -- don't think we need the cert to be executable?

Reply:
It was kept as eXecutable by mistake. I changed it to 0600 on https://github.com/canonical/microceph/pull/355/commits/30356310bbe3e220482f86c56c8ac25227847aca.